File: src/main/kotlin/org/ergoplatform/mosaik/model/ui/layout/HorizontalRule.java
package org.ergoplatform.mosaik.model.ui.layout;

import org.ergoplatform.mosaik.model.Since;
import org.ergoplatform.mosaik.model.ui.ViewElement;

import java.util.Objects;

import javax.annotation.Nonnull;

/**
 * Horizontal Rule displays a horizontal separator or divider
 */
@Since(0)
public class HorizontalRule extends ViewElement {
    @Nonnull
    private Padding vPadding = Padding.DEFAULT;

    @Nonnull
    public Padding getvPadding() {
        return vPadding;
    }

    public void setvPadding(@Nonnull Padding vPadding) {
        Objects.requireNonNull(vPadding);
        this.vPadding = vPadding;
    }
}

File: src/commonMain/kotlin/org/ergoplatform/mosaik/model/ui/layout/HorizontalRule.kt
package org.ergoplatform.mosaik.model.ui.layout

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.ergoplatform.mosaik.model.ui.ViewElement

@Serializable
@SerialName("HorizontalRule")
class HorizontalRule : ViewElement() {
    var vPadding = Padding.DEFAULT
}